src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueue.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Enable.Extensions.Queuing.AzureServiceBus.Internal
+{
+    internal class AzureServiceBusQueue : IDisposable
+    {
+        private int _referenceCount = 1;
+        private bool _disposed = false;
+
+        public AzureServiceBusQueue(string connectionString, string queueName)
+        {
+            Receiver = new MessageReceiver(
+                connectionString,
+                queueName,
+                ReceiveMode.PeekLock);
+
+            Sender = new MessageSender(connectionString, queueName);
+        }
+
+        public IMessageSender Sender { get; }
+        public IMessageReceiver Receiver { get; }
+
+        public int AddReference()
+        {
+            return Interlocked.Increment(ref _referenceCount);
+        }
+
+        public int RemoveReference()
+        {
+            return Interlocked.Decrement(ref _referenceCount);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    Receiver.CloseAsync().GetAwaiter().GetResult();
+                    Sender.CloseAsync().GetAwaiter().GetResult();
```

Comment:
In other `Dispose` methods, we've written the above two lines as:

```
Receiver.CloseAsync()
    .GetAwaiter()
    .GetResult();

Sender.CloseAsync()
    .GetAwaiter()
    .GetResult();
```

Can we do the same here (this is a very minor, stylistic change)?

Reply:
Done.